src/components/map/WorldMap.tsx

```diff
@@ -114,51 +120,67 @@ const WorldMap = () => {
         return polygonString
       }).join()
 
-      const spatialSearchResponse = await fetch('https://graphql.earthdata.nasa.gov/api', {
-        method: 'POST',
-        headers: {
-          Authorization: `Bearer ${authToken}`,
-          "Content-Type": "application/json"
-        },
-        body: JSON.stringify({ query: getGranules, variables: getSpatialSearchGranuleVariables(polygonUrlString, spatialSearchCollectionConceptId, spatialSearchResultLimit) })
-      }).then(async data => {
-        const responseJson = await data.json()
-        // TODO: make subsequent calls to get granules in spatial search area till everything is found.
-
-        const updatedGranules = responseJson.data.granules.items.map((item: any) => {
-          const itemCopy = structuredClone(item)
-          const cpsString = item.granuleUr.match(`${beforeCPS}([0-9]+(_[0-9]+)+)(${afterCPSR}|${afterCPSL})`)?.[1]
-          itemCopy.cpsString = cpsString
-          return itemCopy
-        })
-        const cpsStringTracker: string[] = []
-        const updatedGranulesToUse = updatedGranules.filter((updatedGranuleObject: any) => {
-          // if cpsString not in tracker, it has not been repeated yet. Add to tracker and return
-          const granuleRepeated = cpsStringTracker.includes(updatedGranuleObject.cpsString)
-          if(!granuleRepeated) cpsStringTracker.push(updatedGranuleObject.cpsString)
-          return !granuleRepeated
-          // if cpsString in tracker, it has been repeated. Do not return
-        })
-        const spatialSearchResults = updatedGranulesToUse.map((updatedGranuleObject: any) => {
-          const {producerGranuleId, granuleUr, cpsString, polygons, timeStart, timeEnd} = updatedGranuleObject
-          const cyclePassSceneStringArray = cpsString.split('_').map((id: string) => parseInt(id).toString())
-          const tileValue = parseInt(cyclePassSceneStringArray?.[2] as string)
-          const sceneToUse = String(Math.floor(tileValue))
-          const returnObject: SpatialSearchResult = {
-            cycle: cyclePassSceneStringArray?.[0],
-            pass: cyclePassSceneStringArray?.[1],
-            scene : sceneToUse,
-            producerGranuleId,
-            granuleUr,
-            timeStart,
-            timeEnd,
-            polygons
-          }
-          return returnObject
+      let cursor: string | null = 'initialValue'
+      const spatialSearchItems: any[] = []
+
+      do {
+        if(cursor === 'initialValue') cursor = null
```

Comment:
The do-while will remove the need to set `initialValue` for the cursor

Reply:
Thank you for catching this. I just fixed this and pushed it